modules/light-clients/08-wasm/types/expected_interfaces.go

```diff
@@ -0,0 +1,74 @@
+package types
+
+import (
+	wasmvm "github.com/CosmWasm/wasmvm"
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+)
+
+type WasmEngine interface {
```

Comment:
used same naming convention as [wasmd](https://github.com/CosmWasm/wasmd/blob/894076a25cb2ff5219486e3d3628a5dd982e7ac6/x/wasm/types/wasmer_engine.go#L14) and I only added the interface funcs we use. We could try to add more functions so we could use the other functions without breaking the interface, but I think many of those changes (migrating contracts) require some more substantial architecture and thus would likely be justified to wait for a major release